src/liballoc/rc.rs

```diff
@@ -1161,12 +1161,7 @@ impl<T: ?Sized> RcBoxPtr<T> for Rc<T> {
     #[inline(always)]
     fn inner(&self) -> &RcBox<T> {
         unsafe {
-            // Safe to assume this here, as if it weren't true, we'd be breaking
-            // the contract anyway.
-            // This allows the null check to be elided in the destructor if we
-            // manipulated the reference count in the same function.
-            assume(!(*(&self.ptr as *const _ as *const *const ())).is_null());
```

Comment:
I'm a bit concerned that copying an assume to `NonZero::get` will be disastrous to compile times (as this will show up in all accesses to every collection). Have those bloat problems with assume been resolved?